src/coreclr/vm/siginfo.cpp

```diff
@@ -3908,7 +3952,56 @@ MetaSig::CompareElementType(
                         pOtherModule,
                         tkOther,
                         ClassLoader::ReturnNullIfNotFound,
-                        ClassLoader::FailIfUninstDefOrRef);
+                        ClassLoader::PermitUninstDefOrRef);
+
+                    // If the return type is a generic type definition, we need to
+                    // compare it to the typical method table of the internal type.
+                    if (!hOtherType.IsNull() && hOtherType.IsGenericTypeDefinition())
+                    {
+                        hInternal = TypeHandle{ hInternal.AsMethodTable()->GetTypicalMethodTable() };
+                    }
+
+                    return (hInternal == hOtherType);
+                }
+                case ELEMENT_TYPE_GENERICINST:
+                case ELEMENT_TYPE_BYREF:
```

Comment:
What about arrays and pointers?

For example:
```
unsafe class Test
{
    static void Main()
    {
        MProxy(null, null);
    }

    [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name="M")]
    extern static void MProxy(Test t, [UnsafeAccessorType("Test[][][]")] object p);

    static void M(Test[][][] a) { }
}
```
This fails with MissingMethodException, but even though a matching method exists.

Reply:
Pointers shouldn't matter at present since we require reference types only. I do need to look into arrays though.

Reply:
And this:
```
using System.Runtime.CompilerServices;

#pragma warning disable CS8500

unsafe class Test
{
    static void Main()
    {
        MProxy(null, null);
    }

    [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name="M")]
    extern static void MProxy(Test t, [UnsafeAccessorType("Test")] object* p);

    static void M(Test p) { }
}
```
fails with 

```
Assert failure(PID 10168 [0x000027b8], Thread: 20256 [0x4f20]): Assertion failed 'op1->TypeGet() == TYP_REF' in 'Test:MProxy(Test,ulong)' during 'Importation' (IL size 12; hash 0x3ad4888c; MinOpts)
```